StarTrek13/Xel/borg_structures.dm

```diff
@@ -199,55 +201,168 @@
+
+/obj/machinery/borg/factory/attack_hand(mob/living/carbon/human/user)
+	if(!isborg(user))
+		return
+	var/query
+	query = input(user, "Select an item for production.","Select Production",query) as anything in constructables|"cancel production" // Yes, I know this lists the item paths, not names. It's temporary.
+	if(!query)
+		return
+	if(SSticker.mode.hivemind.cons_points < required_points)
+		to_chat(user, "<span class='warning'>We do not have enough construction materials to fabricate this part.</span>")
+		return
+	if(query == "cancel production")
+		construction(TRUE)
+		update_icon()
+		return
+	to_chat(user, "<span class='info'>The [src] is now producing [query].</span>")
+	constructing = query
+	//part_time = query.cons_time * 10
+	saved_time = world.time + part_time
+	construction()
+	update_icon()
+
+/obj/machinery/borg/factory/proc/construction(var/stop)
+	to_chat(world, "PROCESSING A BORG FACTORY")
+	if(stop)
+		running = FALSE
+		to_chat(world, "PROCESSING A BORG FACTORY STOPPED")
+		//update_icon()
+		return
+	running = TRUE
+	//update_icon()
+	while(world.time < saved_time)
+		sleep(1)
+		to_chat(world, "PROCESSING A BORG FACTORY WHILE")
+	saved_time = world.time + part_time
+	if(!SSticker.mode.hivemind.use_cons_points(required_points))
+		//STOP_PROCESSING(SSobj, src)
+		to_chat(world, "A BORG FACTORY IS MISSING MATERIALS")
+		running = FALSE
+		SSticker.mode.hivemind.message_collective("The [src] at [get_area(src)] has automatically cancelled production. Construction reserves depleted.")
+		update_icon()
+		return
+	new constructing(src.loc)
+	to_chat(world, "BORG FACTORY MADE AN ITEM")
+	construction()
+
+/obj/machinery/borg/converter
+	name = "conversion device"
+	desc = "The final stage of the assimilation process of the borg. May god have mercy on your crew."
+	icon_state = "converter"
+	bound_width = 96
+	layer = 5.5
+	var/running
+	var/inserted_parts = list()
+	var/required_parts = list()
+
+/obj/machinery/borg/converter/update_icon()
+	if(running)
+		icon_state = "converter_active"
+	else
+		icon_state = "converter"
+
+/obj/machinery/borg/converter/attack_hand(mob/living/carbon/human/user) // TODO: MULTISTRUCTURE REQUIREMENT
+	if(running)
+		to_chat(user, "<span class='userdanger'> The device is already running!</span>")
+	if(!isborg(user))
+		to_chat(user, "<span class='warning'>You don't know how to use this.. Yet.</span>")
+		return
+	var/area = get_area(src)
+	var/obj/structure/fluff/helm/desk/tactical/T
+	if(T in area)
+		if(T.theship.assimilated)
+			to_chat(user, "<span class='warning'>This ship has already been assimilated!</span>")
+			return
+		to_chat(user, "<span class='notice'>We begin to initiate assimilation protocols for the [T.theship.name].</span>")
+		if(!do_after(user, 150, target = src))
+			return
+		SSticker.mode.hivemind.message_collective("We have begun assimilation protocols onboard the [T.theship.name].")
+		var/surviving_humans
+		for(var/mob/living/carbon/human/H in area)
```

Comment:
This will get really laggy on highpops. May need a better solution than this

Reply:
how to crash server: click on floor

Reply:
I'll just make it so if someone's starting it, they can't keep starting it
:Thonk: